core/modules/replica/SqlRemoveTablePartitionsJob.h

```diff
@@ -0,0 +1,207 @@
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+#ifndef LSST_QSERV_REPLICA_SQLREMOVETABLEPARTITIONSJOB_H
+#define LSST_QSERV_REPLICA_SQLREMOVETABLEPARTITIONSJOB_H
+
+// System headers
+#include <functional>
+#include <list>
+#include <set>
+#include <string>
+#include <tuple>
+
+// Qserv headers
+#include "replica/SqlJob.h"
+
+// This header declarations
+namespace lsst {
+namespace qserv {
+namespace replica {
+
+/**
+ * Class SqlRemoveTablePartitionsJob represents a tool which will broadcast
+ * the same request for removing MySQL partitions from existing table from all
+ * worker databases of a setup. Result sets are collected in the above defined
+ * data structure.
+ * 
+ * Note, that the algorithm treats regular and partitioned tables quite differently.
+ * For the regular tables it will indeed broadcast exactly the same request
+ * (to the exact table specified as the corresponding parameter of the job)
+ * to all workers. The regular tables must be present at all workers.
+ * The partitioned (chunked) tables will be treated quite differently. First of
+ * all, the name of a table specified as a parameter of the class will be treated
+ * as a class of the tables, and a group of table-specific AND(!) chunk-specific
+ * requests will be generated for such table. For example, of the table name is:
+ *
+ *   'Object'
+ * 
+ * and the following table replicas existed for the table at a time of the request:
+ * 
+ *    worker | chunk
+ *   --------+-----------------------
+ *      A    |  123
+ *   --------+-----------------------
+ *      B    |  234
+ *   --------+-----------------------
+ *      C    |  234
+ *      D    |  345
+ *
+ * then the low-level requests will be sent for the following tables to
+ * the corresponding workers:
+ * 
+ *    worker | table
+ *   --------+-----------------------
+ *      A    | Object
+ *      A    | Object_123
+ *      A    | ObjectFullOverlap_123
+ *   --------+-----------------------
+ *      B    | Object
+ *      B    | Object_234
+ *      B    | ObjectFullOverlap_234
+ *   --------+-----------------------
+ *      C    | Object
+ *      C    | Object_234
+ *      C    | ObjectFullOverlap_234
+ *   --------+-----------------------
+ *      D    | Object
+ *      D    | Object_345
+ *      D    | ObjectFullOverlap_345
```

Comment:
This looks more complicate than it should be. Why can't you send "remove all chunks (and overlaps)" request to a worker?

Reply:
This may be a good suggestion! Thank you! I will consider this optimization.

Reply:
For the record, this suggestion was registered at:
https://confluence.lsstcorp.org/display/DM/Completed%2C+on-going+and+proposed+improvements
I'm going to defer it till after I finish with publishing the initial implementation of the Ingest system.